File: client/old/src/main/java/net/treset/minecraftlauncher/util/string/TimeString.java
package net.treset.minecraftlauncher.util.string;

import net.treset.minecraftlauncher.LauncherApplication;

public class TimeString extends FormatString {
    private final String output;

    public TimeString(long seconds) {
        if (seconds < 60) {
            this.output = seconds + LauncherApplication.stringLocalizer.get("suffix.seconds");
        } else if (seconds < 60 * 60) {
            this.output = (seconds / 60) + LauncherApplication.stringLocalizer.get("suffix.minutes");
        } else if (seconds < 60 * 60 * 10) {
            this.output = (seconds / 3600) + LauncherApplication.stringLocalizer.get("suffix.hours") + " " + ((seconds % 3600) / 60) + LauncherApplication.stringLocalizer.get("suffix.minutes");
        } else if (seconds < 60 * 60 * 24) {
            this.output = (seconds / 3600) + LauncherApplication.stringLocalizer.get("suffix.hours");
        } else if (seconds < 60 * 60 * 24 * 10) {
            this.output = (seconds / (3600 * 24)) + LauncherApplication.stringLocalizer.get("suffix.days") + " " + ((seconds % (3600 * 24)) / 3600) + LauncherApplication.stringLocalizer.get("suffix.hours");
        } else {
            this.output =  (seconds / (3600 * 24)) + LauncherApplication.stringLocalizer.get("suffix.days");
        }
    }

    public String get() {
        return this.output;
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/string/TimeString.kt
package net.treset.treelauncher.backend.util.string

import net.treset.treelauncher.localization.strings

class TimeString(seconds: Long) : FormatString() {
    private var output: String

    init {
        output =
        if (seconds < 60) {
            "${seconds}${strings().units.seconds()}"
        } else if (seconds < 60 * 60) {
            "${seconds / 60}${strings().units.minutes()}"
        } else if (seconds < 60 * 60 * 10) {
            "${seconds / 3600}${strings().units.hours()} ${seconds % 3600 / 60}${strings().units.minutes()}"
        } else if (seconds < 60 * 60 * 24) {
            "${seconds / 3600}${strings().units.hours()}"
        } else if (seconds < 60 * 60 * 24 * 10) {
            "${seconds / (3600 * 24)}${strings().units.days()} ${seconds % (3600 * 24) / 3600}${strings().units.hours()}"
        } else {
            "${seconds / (3600 * 24)}${strings().units.days()}"
        }
    }

    override fun get(): String {
        return output
    }
}
